Sources/SWBCore/Settings/Settings.swift

```diff
@@ -1265,6 +1265,35 @@ private class SettingsBuilder: ProjectMatchLookup {
         }
     }
 
+    enum FindPlatformError: Error { case message(String) }
+
+    func findPlatform(for triple: String, core: Core) -> Result<Platform,FindPlatformError> {
+        let llvmTriple = try? LLVMTriple(triple)
+
+        guard let llvmTriple else {
+            return .failure(.message("invalid triple \(triple)"))
+        }
+
+        var platformNames = OrderedSet<String>()
+        for platformExtension in core.pluginManager.extensions(of: PlatformInfoExtensionPoint.self) {
+            if let platformName = platformExtension.platformName(triple: llvmTriple) {
+                platformNames.append(platformName)
+            }
+        }
+
+        guard let platformName = platformNames.first, platformNames.count == 1 else {
```

Comment:
`only` is the idiomatic way to do this, which handles checking for exactly one value and returning it.

```suggestion
        guard let platformName = platformNames.only else {
```